apps/emqx_durable_storage/src/emqx_ds.erl

```diff
@@ -230,12 +242,36 @@ update_iterator(DB, OldIter, DSKey) ->
 
 -spec next(db(), iterator(), pos_integer()) -> next_result().
 next(DB, Iter, BatchSize) ->
-    ?module(DB):next(DB, Iter, BatchSize).
+    Opts = #{use_cache => false},
+    next(DB, Iter, BatchSize, Opts).
+
+-spec next(db(), iterator(), pos_integer(), next_opts()) -> next_result().
+next(DB, Iter, BatchSize, Opts) ->
+    ?module(DB):next(DB, Iter, BatchSize, Opts).
```

Comment:
Seems that the caching is done on the replication layer, i.e. inside the concrete `emqx_ds` implementation. However, the caching options are raised to the `emqx_ds` interface. 

So, on the one hand, the implementations should be aware of this option; on the other hand, they cannot use the caching bc. it is replication backend-specific.

So maybe we should
* Either lift caching to the `emqx_ds` level, so that other backends could get it for free;
* Or remove caching from the `emqx_ds` interface, so that each backend opaquely implements caching (if needed)
?

Reply:
Or is this the intended methos of integration? 🤔
Should the other backends, like replication layer, query the "global" cache themselves, when use_cache is true? And query the db otherwise?

Reply:
> Either lift caching to the emqx_ds level, so that other backends could get it for free;

The cache relies on `last_seen_key` being implemented to work properly, so I think it would be best left for each backend to do similar wrapping :sweat_smile: 

WDYT?